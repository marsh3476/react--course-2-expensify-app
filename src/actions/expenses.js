import { v4 as uuidv4 } from 'uuid';

// ADD EXPENSE
export const addExpense = (
  { 
    description = '', 
    note = '', 
    amount = 0, 
    createdAt=0 
  } = {}
) => ({
  type: 'ADD_EXPENSE',
  expense: {
    id: uuidv4(),
    description: description,
    note: note, 
    amount: amount,
    createdAt: createdAt
  }
});

// EDIT EXPENSE
export const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id: id,
  updates: updates
});
// REMOVE EXPENSE
export const removeExpense = ({ id } = {}) => {
  return {
    type: 'REMOVE_EXPENSE',
    id: id
  }
  
};