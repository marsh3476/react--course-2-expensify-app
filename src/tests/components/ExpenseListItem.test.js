import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';


test('should create Expense List Item component', () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} key={1} />);
  expect(wrapper).toMatchSnapshot();
})