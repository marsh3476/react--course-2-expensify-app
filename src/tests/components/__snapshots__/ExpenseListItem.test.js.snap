// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create Expense List Item component 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    Amount: $
    195
  </p>
  <p>
    Created At: 
    0
  </p>
</div>
`;
